src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -447,178 +290,171 @@ static smart_amp_proc get_smart_amp_process(struct comp_dev *dev,
 	}
 }
 
-static int smart_amp_copy(struct comp_dev *dev)
+/* Only reset the src_input and fb_input if the num_input_buffers
+ * (get from module adapter)has changed.
+ */
+static void smart_amp_input_setup(struct processing_module *mod,
+				  struct input_stream_buffer *inputs,
+				  int32_t num_input_buffers)
 {
-	struct smart_amp_data *sad = comp_get_drvdata(dev);
-	struct comp_buffer __sparse_cache *source_buf = buffer_acquire(sad->source_buf);
-	struct comp_buffer __sparse_cache *sink_buf = buffer_acquire(sad->sink_buf);
-	uint32_t avail_passthrough_frames;
-	uint32_t avail_feedback_frames;
-	uint32_t avail_frames = 0;
-	uint32_t source_bytes;
-	uint32_t sink_bytes;
-	uint32_t feedback_bytes;
+	struct smart_amp_data *sad = module_get_private_data(mod);
+	struct comp_buffer *buf;
+	struct comp_buffer __sparse_cache *buffer_c;
+	int i;
 
-	comp_dbg(dev, "smart_amp_copy()");
+	if (sad->input_stream_num == num_input_buffers)
+		return;
+
+	if (num_input_buffers == 1) {
+		sad->src_input = &inputs[0];
+		sad->src_buf = attr_container_of(inputs[0].data,
+						 struct comp_buffer __sparse_cache,
+						 stream, __sparse_cache);
+		sad->fb_input = NULL;
+		sad->fb_buf = NULL;
```

Comment:
can we set the sad->input_stream_num at the beginning and return here? Then we can reduce the indentation after removing the else

Reply:
done.